Fix an endian problem in upb that caused protoc plugins to crash on big endian
platforms (s390x). Closed enum fields skip the int32 munge, which is harmless on
little-endian but decodes as 0 on big-endian, so protoc-gen-upb* segfaults while
building Bazel.

Upstream fix, first released in protobuf 35.0:
https://github.com/protocolbuffers/protobuf/commit/8282f0f8e (#25363)

--- a/upb/wire/decode.c
+++ b/upb/wire/decode.c
@@ -1079,6 +1079,7 @@
           *op = kUpb_DecodeOp_UnknownField;
           return ptr;
         }
+        _upb_Decoder_MungeInt32(val);
       } else {
         _upb_Decoder_Munge(field, val);
       }
